src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -99,7 +105,20 @@ public function getArguments(Request $request, callable $controller, ?\Reflectio
                 }
             }
 
-            throw new \RuntimeException(sprintf('Controller "%s" requires that you provide a value for the "$%s" argument. Either the argument is nullable and no null value has been provided, no default value has been provided or there is a non-optional argument after this one.', $this->getPrettyName($controller), $metadata->getName()));
+            $reasons = array_map(static fn (NearMissValueResolverException $e) => $e->getMessage(), $valueResolverExceptions);
+            if (empty($reasons)) {
+                $reasons[] = 'Either the argument is nullable and no null value has been provided, no default value has been provided or there is a non-optional argument after this one.';
+            }
+
+            $reasonCounter = 1;
+            if (\count($reasons) > 1) {
+                foreach ($reasons as $i => $reason) {
+                    $reasons[$i] = $reasonCounter.') '.$reason;
+                    ++$reasonCounter;
+                }
+            }
```

Comment:
```suggestion
            } elseif (\count($reasons) > 1) {
                $reasonCounter = 1;
                foreach ($reasons as $i => $reason) {
                    $reasons[$i] = $reasonCounter.') '.$reason;
                    ++$reasonCounter;
                }
            }
```

Reply:
It is used at line 121 so have to be initialized outside `if` block